llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -297,7 +298,16 @@ def run_event(self, event: Event, state: State, logger: Logger) -> None:
                     + f'Got {type(state.model)} instead.',
                 )
             if self.remote_ud is not None:
-                self.remote_ud.init(state, logger)
+                try:
+                    self.remote_ud.init(state, logger)
+                except PermissionError as e:
+                    if 'Client Error' in str(
+                        e,
+                    ):  # thrown from composer.utils._wrap_mlflow_exceptions
+                        raise StoragePermissionError(
+                            'Error when write to save_folder.',
```

Comment:
do you want to propagate the error message up so that we can at least log the full error in mcloud?
```suggestion
                            f'Error when write to save_folder. Details: {e.message}',
```

Reply:
I think it's fine as-is since the earlier error will appear in the stacktrace and logs